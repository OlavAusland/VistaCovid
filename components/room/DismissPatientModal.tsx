import { Modal, Pressable, Text, View } from "react-native";
import { adminStyle } from "../../styles/AdminStyles";

type DismissPatientModal={
    modalVisible: boolean;
    handleRequestClose: Function;
    dismissAndClose: Function;
}

export const DismissPatientModal = (props: DismissPatientModal) => {
    
    return(
        <Modal
        animationType="slide"
        statusBarTranslucent={true}
        transparent={true}
        visible={props.modalVisible}
        onRequestClose={()=>props.handleRequestClose()}>

                <View style={adminStyle.centeredView}>
                    <View style={adminStyle.modalView}>
                        <Text style={adminStyle.modalText}>Are you sure you want to dismiss patient?</Text>
                        
                        <Pressable
                        style={[adminStyle.button, adminStyle.buttonOpen]}
                        onPress={()=> props.dismissAndClose()}
                        >
                            <Text style={adminStyle.textStyle}>Yes, dismiss.</Text>
                        </Pressable>

                        <Pressable
                        style={[adminStyle.button, adminStyle.buttonClose]}
                        onPress={()=> props.handleRequestClose()}
                        >
                            <Text style={adminStyle.textStyle}>No, close.</Text>
                        </Pressable>
                    </View>
                </View>
            </Modal>
    );
}

