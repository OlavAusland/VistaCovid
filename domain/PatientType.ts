export type Patient1 = {
    firstname: string,
    lastname: string,
    ssn: string,
    phone: string | undefined,
    relatives: string | undefined,
    address: string | undefined,
        city: string | undefined,
    journals: Array<Object> | undefined
}

export type Patient = {
    firstname: string,
    lastname: string,
    ssn: string,
    gender: string,
    address: string,
    housenumber:string,
    city: string,
    coAddress: string,
    gradering: string
}