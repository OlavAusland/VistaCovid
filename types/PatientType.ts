export type Patient = {
    firstname: string,
    lastname: string,
    ssn: string,
    phone: string | undefined,
    relatives: string | undefined,
    address: string | undefined,
        city: string | undefined,
        journals: Array<Object> | undefined
}