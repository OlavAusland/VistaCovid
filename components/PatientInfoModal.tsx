import { Text, View, TextInput, Pressable, Modal, ScrollView } from 'react-native';
import React, { useEffect, useState } from 'react';
import { getPatient } from '../api/folkeregisterModelAPI';
import { FolkeregisterPerson } from '../domain/PatientType';
import { patientInfoStyle } from '../styles/PatientInfoStyle';


type PatientInfoModalProps = {
    modalVisible: boolean;
    handleRequestClose: Function;
    fnr: string;
}


export const PatientInfoModal = (props: PatientInfoModalProps) => {
    const [patient, setPatient] = useState<FolkeregisterPerson>();

    useEffect(() => {
        getPatient(props.fnr).then(patient => {
            setPatient(patient);
        }).catch(err => { console.log(err) });
    }, []);

    return (

        <Modal
            animationType="slide"
            statusBarTranslucent={true}
            transparent={true}
            visible={props.modalVisible}
            onRequestClose={() => props.handleRequestClose()}
        >
            <View style={{ top: '20%', bottom: '15%' }}>
                <View style={patientInfoStyle.container}>
                    <Text style={{ fontSize: 30, fontWeight: '200', marginTop: 10, marginBottom: 10 }}>Patient</Text>
                    <View style={patientInfoStyle.text}>
                        <Text style={{ fontSize: 15, marginBottom: 10 }}>
                            <Text style={{ fontWeight: 'bold' }}>Lastname: </Text>
                            <Text>{patient?.lastname}</Text>
                        </Text>
                        <Text style={{ fontSize: 15, marginBottom: 10 }}>
                            <Text style={{ fontWeight: 'bold' }}>Firstname: </Text>
                            <Text>{patient?.firstname} {patient?.midlename}</Text>
                        </Text>
                        <Text style={{ fontSize: 15, marginBottom: 10 }}>
                            <Text style={{ fontWeight: 'bold' }}>SSN: </Text>
                            <Text>{patient?.ssn}</Text>
                        </Text>
                        <Text style={{ fontSize: 15, marginBottom: 10 }}>
                            <Text style={{ fontWeight: 'bold' }}>Gender: </Text>
                            <Text>{patient?.gender}</Text>
                        </Text>
                        <Text style={{ fontSize: 15, marginBottom: 10 }}>
                            <Text style={{ fontWeight: 'bold' }}>Address: </Text>
                            <Text>{patient?.address} {patient?.housenumber}</Text>
                        </Text>
                        <Text style={{ fontSize: 15, marginBottom: 10 }}>
                            <Text style={{ fontWeight: 'bold' }}>City: </Text>
                            <Text>{patient?.city}</Text>
                        </Text>
                        <Text style={{ fontSize: 15, marginBottom: 10 }}>
                            <Text style={{ fontWeight: 'bold' }}>Co-Address: </Text>
                            <Text>{patient?.coAddress}</Text>
                        </Text>
                    </View>
                    <Pressable
                        onPress={() => props.handleRequestClose()} >
                        <View style={patientInfoStyle.button}>
                            <Text style={{ color: 'white', alignSelf: 'center', fontSize: 20 }}>Close</Text>
                        </View>
                    </Pressable>
                </View>
            </View>
        </Modal>
    );
}

