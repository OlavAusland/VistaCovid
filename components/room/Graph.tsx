import { Dimensions, Platform, TouchableOpacity } from 'react-native';
import { LineChart } from 'react-native-chart-kit';
import { GraphData } from '../../domain/RoomType';

export type GraphProps = {
    data: GraphData[] | undefined,
    name: string,
    color: string
}

export const LineGraph = (props: GraphProps) => {

    const xData = props.data?.map((res: GraphData) => {return res.time.toString()});
    const yData = props.data?.map((res: GraphData) => {return res.value});

    return(
        <TouchableOpacity>
            <LineChart
            data={{
                labels: [],
                datasets: [
                    {
                        data: yData? yData.slice(-15) : [],
                    }
                ],
                legend: [props.name]
            }}
            width={Platform.OS === 'android' ? Dimensions.get('window').width / 1.1 : 375} // from react-native
            height={200}
            xAxisLabel='s'
            yAxisInterval={1} // optional, defaults to 1
            chartConfig={{
                useShadowColorFromDataset:true,
                fillShadowGradientOpacity:0.5,
                backgroundGradientFrom: props.color,
                backgroundGradientTo: props.color,
                decimalPlaces: 1, // optional, defaults to 2dp
                color: () => '#C1E8FD',
                labelColor: (opacity = 1) => `rgba(0, 0, 0, ${opacity})`,
                style: {
                    borderRadius: 16
            },
            propsForDots: {
                r: "2",
                strokeWidth: "4",
                stroke: "#000000"
            }
            }}
            style={{
                marginVertical: 8,
                borderRadius: 16,
            }}
            fromZero={false}
        />
        </TouchableOpacity>
    )
}
