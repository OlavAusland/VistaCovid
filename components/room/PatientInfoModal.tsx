import { Text, View, TextInput, Pressable, Modal, ScrollView } from 'react-native';
import { useEffect, useState } from 'react';
import { getPatient as folkeregisterGetPatient} from '../../api/folkeregisterModelAPI';
import {getPatient as firebaseGetPatient} from '../../api/firebaseAPI';
import { FolkeregisterPerson, Patient } from '../../domain/PatientType';
import { patientInfoStyle } from '../../styles/PatientInfoStyle';
import { PatientInfoModalProps } from '../../domain/patientInfoType';


export const PatientInfoModal = (props: PatientInfoModalProps) => {
    const [patient, setPatient] = useState<FolkeregisterPerson | Patient>();
    const [error, setError] = useState<Error | undefined>(undefined);

    useEffect(() => {
        folkeregisterGetPatient(props.fnr).then(patient => {
            setPatient(patient);
        }).catch((err) =>{setError(err);});

        if(!patient){
            firebaseGetPatient(props.fnr).then(patient => {
                setPatient(patient);
            }).catch((err) =>{setError(err);});
        }
        
    }, []);

    useEffect(() => {
        console.log(patient);
    }, [patient]);

/* 
    if (error) {
        return <View>
            <Text>{error.message}</Text>
            <Text>Can't find patient info</Text>
        </View>
    } */

    return (

        <Modal
            animationType="slide"
            statusBarTranslucent={true}
            transparent={true}
            visible={props.modalVisible}
            onRequestClose={() => props.handleRequestClose()}
            testID="patientInfoModal"
        >
            <View style={{ top: '20%', bottom: '15%' }}>
                <View style={patientInfoStyle.container}>
                    <Text style={{ fontSize: 30, fontWeight: '200', marginTop: 10, marginBottom: 10 }}>Patient</Text>
                    <View style={patientInfoStyle.text}>
                        <Text style={{ fontSize: 15, marginBottom: 10 }}>
                            <Text style={{ fontWeight: 'bold' }}>Lastname: </Text>
                            <Text testID='lastname'>{patient?.lastname}</Text>
                        </Text>
                        <Text style={{ fontSize: 15, marginBottom: 10 }}>
                            <Text style={{ fontWeight: 'bold' }}>Firstname: </Text>
                            <Text testID='firstname'>{patient?.firstname} {patient?.midlename}</Text>
                        </Text>
                        <Text style={{ fontSize: 15, marginBottom: 10 }}>
                            <Text style={{ fontWeight: 'bold' }}>SSN: </Text>
                            <Text testID='ssn'>{patient?.ssn}</Text>
                        </Text>
                        <Text style={{ fontSize: 15, marginBottom: 10 }}>
                            <Text style={{ fontWeight: 'bold' }}>Gender: </Text>
                            <Text testID='gender'>{patient?.gender}</Text>
                        </Text>
                        <Text style={{ fontSize: 15, marginBottom: 10 }}>
                            <Text style={{ fontWeight: 'bold' }}>Address: </Text>
                            <Text  testID='address'>{patient?.address} {patient?.housenumber}</Text>
                        </Text>
                        <Text style={{ fontSize: 15, marginBottom: 10 }}>
                            <Text style={{ fontWeight: 'bold' }}>City: </Text>
                            <Text  testID='city'>{patient?.city}</Text>
                        </Text>
                        <Text style={{ fontSize: 15, marginBottom: 10 }}>
                            <Text style={{ fontWeight: 'bold' }}>Co-Address: </Text>
                            <Text  testID='co.adress'>{patient?.coAddress}</Text>
                        </Text>
                    </View>
                    <Pressable
                        onPress={() => props.handleRequestClose()} >
                        <View style={patientInfoStyle.button}>
                            <Text style={{ color: 'white', alignSelf: 'center', fontSize: 20 }}>Close</Text>
                        </View>
                    </Pressable>
                </View>
            </View>
        </Modal>
    );
}

