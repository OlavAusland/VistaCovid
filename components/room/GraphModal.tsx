import { Modal, Button, Pressable, ScrollView, Text, TouchableOpacity, View} from "react-native";
import { graphModalStyle } from "../../styles/GraphModalStyles";
import { LineGraph } from "./Graph";
import { GraphData} from "../../types/RoomType";
import { useState } from "react";

import DateTimePickerModal from "react-native-modal-datetime-picker";

export type GraphModalTypes = {
    data: GraphData[],
    setModal: (modal: boolean) => void,
    modal: boolean
}


export const GraphModal = (props: GraphModalTypes) => {
    const [date, setDate] = useState({startDate:{date: new Date(), visible: false}, endDate:{date: new Date(), visible: false}});

    

    return (
        <Modal  
            animationType="slide"
            transparent={false}
            visible={props.modal}
            onRequestClose={() => {
                props.setModal(false);
            }}
        >
            <ScrollView contentContainerStyle={graphModalStyle.container}>
                <LineGraph data={[{time:0, value:60}]} name={'Heart Rate'}
                            modal={props.modal} setModal={props.setModal}/>
                <View style={{flexDirection:'row', width:'90%', justifyContent:'center'}}>
                    <View style={{flex:1, justifyContent:'center', alignItems:'center', backgroundColor:'#0274A1', paddingTop:10, paddingBottom:10}}>
                        <Button title="Set Start Date" onPress={() => {setDate({...date, startDate:{...date.startDate, visible: true}})}}/>
                        <DateTimePickerModal
                            isVisible={date.startDate.visible}
                            date={date.startDate.date}
                            mode="date"
                            onConfirm={(newDate: Date) => {setDate({...date, startDate:{date:newDate, visible: false}})}}
                            onCancel={() => {setDate({...date, startDate:{...date.startDate, visible: false}})}}
                        />
                    </View>
                    <View style={{flex:1, justifyContent:'center', alignItems:'center', backgroundColor:'#0274A1', paddingTop:10, paddingBottom:10}}>
                        <Button title="Set End Date" onPress={() => {setDate({...date, endDate:{...date.endDate, visible: true}})}}/>
                        <DateTimePickerModal
                            isVisible={date.endDate.visible}
                            date={date.endDate.date}
                            mode="date"
                            onConfirm={(newDate: Date) => {setDate({...date, endDate:{date:newDate, visible: false}})}}
                            onCancel={() => {setDate({...date, endDate:{...date.endDate, visible: false}})}}
                        />
                    </View>
                </View>
                <View style={graphModalStyle.notes}>
                    <Text></Text>
                </View>
                <TouchableOpacity onPress={() => {props.setModal(false)}}>
                    <Text>Exit</Text>
                </TouchableOpacity>
            </ScrollView>
        </Modal>
    );
}