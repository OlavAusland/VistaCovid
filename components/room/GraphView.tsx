import { ScrollView, View } from 'react-native';
import { Room } from '../../domain/RoomType';
import { roomStyle } from '../../styles/RoomStyles';
import { LineGraph } from './Graph';

export type GraphViewProps = {
    room: Room,
    setModal: (modal: boolean) => void,
    modal: boolean
}

export  const GraphView = (props: GraphViewProps) => {
    return (
        <ScrollView contentContainerStyle={roomStyle.body}>
            <View style={roomStyle.graphContainer}>
                <LineGraph color={'#eb4034'} data={props.room?.heartRate} name={'Heart Rate'}
                            modal={props.modal} setModal={props.setModal}/>
            </View>
            <View style={roomStyle.graphContainer}>
                <LineGraph color={'#edb855'} data={props.room?.bloodPressure} name={'Blood Pressure'}
                            modal={props.modal} setModal={props.setModal}/>
            </View>
            <View style={roomStyle.graphContainer}>
                <LineGraph color={'#6ed7e0'} data={props.room?.oxygenLevel}  name={'Oxygen Level'}
                            modal={props.modal} setModal={props.setModal}/>
            </View>
        </ScrollView>
    );
}