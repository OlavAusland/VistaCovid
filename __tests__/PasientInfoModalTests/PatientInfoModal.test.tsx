
import { act, render } from '@testing-library/react-native';
import { PatientInfoModal } from "../../components/room/PatientInfoModal";
import { FolkeregisterPerson } from '../../domain/PatientType';

const folkeregisterPerson: FolkeregisterPerson = {
    firstname: 'Siri',
    midlename: 'Hanne',
    lastname: 'Olsen',
    ssn: '30070123456',
    gender: 'kvinne',
    address: 'maurliveien',
    housenumber: '1',
    city: 'Oslo',
    coAddress: '',
    gradering: 'ugradert'
}



// Snapshot test
test("renders correctly", () => {
    const tree = render(<PatientInfoModal handleRequestClose={jest.fn()} patient={folkeregisterPerson} />).toJSON();
        expect(tree).toMatchSnapshot();
});

//Unit tests
/* 
const stubbedPatient = [
    {
        person: {
            personnavn: [
                {
                    fornavn: "Siri",
                    mellomnavn: "Hanne",
                    etternavn: "Olsen",
                },
            ],
            bostedsadresse: [
                {
                    vegadresse: {
                        kommunenummer: "0301",
                        bruksenhetsnummer: "0101",
                        bruksenhetstype: "bolig",
                        adressenavn: "maurliveien",
                        adressenummer: "1",
                        adressekode: "",
                        adressetileggsnavn: "",
                        poststed: "Oslo",
                        coAdressenavn: "",
                    },
                },
            ],
            adressebeskyttelse: [
                {
                    graderingsnivå: "ugradert",
                },
            ],
            kjønn: [
                {
                    kjønn: "kvinne",
                },
            ],
            identifikasjonsnummer: [
                {
                    status: "iBruk",
                    fødselsEllerDNnummer: "30070123456",
                    identifikatortype: "fødselsnummer",
                },
            ],
        },
    },
]; */


test("shows patient when data are fetched", async () => {

    /* const getPatient = jest.fn().mockReturnValueOnce({
        loading: false,
        data: stubbedPatient,
        error: null,
    }); */

    const { getByTestId } = render(<PatientInfoModal handleRequestClose={jest.fn()} patient={folkeregisterPerson}/>);
    expect(getByTestId("patientInfoModal")).not.toBeNull();

    const lastnametest = getByTestId("lastname") 
    await act(async () => {
        expect(lastnametest.props.children).toContain("Olsen");
      });

    const firstnametest1 = getByTestId("firstname") 
    await act(async () => {
        expect(firstnametest1.props.children[0]).toContain("Siri" );
    });

    const firstnametest2 = getByTestId("firstname") 
    await act(async () => {
        expect(firstnametest2.props.children[2]).toContain("Hanne");
    });
        
    const ssntest = getByTestId("ssn") 
    await act(async () => {
        expect(ssntest.props.children).toContain("30070123456");
    });

    const gendertest = getByTestId("gender") 
    await act(async () => {
        expect(gendertest.props.children).toContain("kvinne");
    });

    const addresstest1 = getByTestId("address") 
    await act(async () => {
        expect(addresstest1.props.children[0]).toContain("maurliveien");
    });
    
    const addresstest2 = getByTestId("address") 
    await act(async () => {
        expect(addresstest2.props.children[2]).toContain("1");
    });

    const citytest = getByTestId("city") 
    await act(async () => {
        expect(citytest.props.children).toContain("Oslo");
    });

    const coaddress = getByTestId("co.address") 
    await act(async () => {
        expect(coaddress.props.children).toContain("");
    });

});





