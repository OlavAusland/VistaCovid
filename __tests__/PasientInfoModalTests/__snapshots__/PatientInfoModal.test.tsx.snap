// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Modal
  animationType="slide"
  hardwareAccelerated={false}
  onRequestClose={[Function]}
  statusBarTranslucent={true}
  testID="patientInfoModal"
  transparent={true}
  visible={true}
>
  <View
    style={
      Object {
        "bottom": "15%",
        "top": "20%",
      }
    }
  >
    <View
      style={
        Object {
          "alignContent": "flex-end",
          "alignSelf": "flex-end",
          "backgroundColor": "#9dd4fb",
          "elevation": 5,
          "padding": 20,
          "paddingLeft": 20,
          "paddingRight": 20,
          "width": "50%",
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 30,
            "fontWeight": "200",
            "marginBottom": 10,
            "marginTop": 10,
          }
        }
      >
        Patient
      </Text>
      <View
        style={
          Object {
            "backgroundColor": "white",
            "padding": 10,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 15,
              "marginBottom": 10,
            }
          }
        >
          <Text
            style={
              Object {
                "fontWeight": "bold",
              }
            }
          >
            Lastname: 
          </Text>
          <Text
            testID="lastname"
          />
        </Text>
        <Text
          style={
            Object {
              "fontSize": 15,
              "marginBottom": 10,
            }
          }
        >
          <Text
            style={
              Object {
                "fontWeight": "bold",
              }
            }
          >
            Firstname: 
          </Text>
          <Text>
             
          </Text>
        </Text>
        <Text
          style={
            Object {
              "fontSize": 15,
              "marginBottom": 10,
            }
          }
        >
          <Text
            style={
              Object {
                "fontWeight": "bold",
              }
            }
          >
            SSN: 
          </Text>
          <Text />
        </Text>
        <Text
          style={
            Object {
              "fontSize": 15,
              "marginBottom": 10,
            }
          }
        >
          <Text
            style={
              Object {
                "fontWeight": "bold",
              }
            }
          >
            Gender: 
          </Text>
          <Text />
        </Text>
        <Text
          style={
            Object {
              "fontSize": 15,
              "marginBottom": 10,
            }
          }
        >
          <Text
            style={
              Object {
                "fontWeight": "bold",
              }
            }
          >
            Address: 
          </Text>
          <Text>
             
          </Text>
        </Text>
        <Text
          style={
            Object {
              "fontSize": 15,
              "marginBottom": 10,
            }
          }
        >
          <Text
            style={
              Object {
                "fontWeight": "bold",
              }
            }
          >
            City: 
          </Text>
          <Text />
        </Text>
        <Text
          style={
            Object {
              "fontSize": 15,
              "marginBottom": 10,
            }
          }
        >
          <Text
            style={
              Object {
                "fontWeight": "bold",
              }
            }
          >
            Co-Address: 
          </Text>
          <Text />
        </Text>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
      >
        <View
          style={
            Object {
              "alignSelf": "center",
              "backgroundColor": "#0274a1",
              "borderRadius": 10,
              "height": 40,
              "justifyContent": "center",
              "marginTop": 30,
              "width": "25%",
            }
          }
        >
          <Text
            style={
              Object {
                "alignSelf": "center",
                "color": "white",
                "fontSize": 20,
              }
            }
          >
            Close
          </Text>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;
